processing/src/main/java/io/druid/query/metadata/metadata/SegmentMetadataQuery.java

```diff
@@ -121,11 +117,17 @@ public SegmentMetadataQuery(
     if (querySegmentSpec == null) {
       this.usingDefaultInterval = true;
     } else {
-      this.usingDefaultInterval = useDefaultInterval == null ? false : useDefaultInterval;
+      if (querySegmentSpec.getIntervals().size() == 1 && querySegmentSpec.getIntervals()
```

Comment:
Ok, I somehow missed `.get(0)` on the next line. The formatting is strange: suggested
```java
if (querySegmentSpec.getIntervals().size() == 1 &&
    querySegmentSpec.getIntervals().get(0).equals(DEFAULT_INTERVAL)) {
```